### DISCUSSION TITLE
How do I replicate the 'conventions' icons used in the Material-for-mkdocs documentation site?

### DISCUSSION THREAD

**Discussion by rathboma** [squidfunk/mkdocs-material#6544] [Q&A] [answered]
Hi everyone,

First, wow is this an amazing project. The docs produced are fantastic.

Quick question - I'd like to replicate the icons used in the project documentation, like these:

![image](https://github.com/squidfunk/mkdocs-material/assets/279769/7e2fc5c4-309a-47a4-8049-8ed3e5c03cc3)

I can see in the code that they use this block to achieve this:

```html
<!-- md:version 9.2.0 -->
<!-- md:plugin -->
<!-- md:flag experimental -->
```

I can't figure out if this is a built in feature? How would I achieve the same thing?

Thanks anyone for your help!

**Comment by kamilkrzyskow** [comment, marked as answer]
Hello @rathboma,
this was asked many times in different ways in the discussions, searching with `conventions` I got to this discussion: 
- https://github.com/squidfunk/mkdocs-material/discussions/6045

To answer in short again:

0. This is currently a custom feature of the mkdocs-material docs, and isn't exported in a straightforward way for other users to use. However, this will likely change, as people often ask for it.
1. The `<!-- md:plugin -->` tokens/identifiers/shortcodes are processed by the shortcodes hook:
    https://github.com/squidfunk/mkdocs-material/blob/master/src/overrides/hooks/shortcodes.py
2. The hook replaces the shortcodes with a Markdown structure
3. The HTML is styled additionally with a custom CSS (mkdocs-material uses SCSS, which is later compiled and minified): 
https://github.com/squidfunk/mkdocs-material/blob/175df613970869744114e0c4299f0e448aba8737/src/overrides/assets/stylesheets/custom/_typeset.scss#L97

To enable the hook, you need to add it to the `mkdocs.yml` file, as described here:
https://www.mkdocs.org/user-guide/configuration/#hooks

**Reply by rathboma** [reply]
Thanks for the response!

**Reply by squidfunk** [reply]
In the future, we'll evaluate whether we can somehow generalize the shortcodes, but for now, copying and adapting our hook is the best solution. Please understand that we must prioritize what we work on, and our current focus is tags, projects and search. However, we're planning to do a face lift in 2024 and make Material for MkDocs even more easily customizable.

**Reply by axololly** [reply]
@squidfunk Are there any plugins that will enable you to use these without needing to rewrite `shortcodes.py` and the CSS involved? I can't seem to find any online, no matter where I look.

**Reply by squidfunk** [reply]
None that I know of, but I guess it's a good case for a third party plugin. We can't offer it as built-in functionality, because it needs too much orchestration – it's not a mere configuration issue, it would need to be a plugin. Thus, I'd invite you to reuse our code and move it into a fully fledged plugin ☺️

**Reply by axololly** [reply]
Hm, alright. What extension do you use for the previews of dot-underlined hyperlinks?

For reference, I'm talking about:
- the "Caret & Tilde" hyperlink [here](https://squidfunk.github.io/mkdocs-material/reference/formatting/#sub-and-superscripts)
- the "Keys" hyperlink [here](https://squidfunk.github.io/mkdocs-material/reference/formatting/#adding-keyboard-keys)

**Reply by squidfunk** [reply]
That'll be [instant previews](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#instant-previews). Please note that this feature is currently [reserved to sponsors](https://github.com/sponsors/squidfunk) and distributed as part of [Insiders](https://squidfunk.github.io/mkdocs-material/insiders/).

**Reply by axololly** [reply]
Ah, okay. Since it's currently not available to the public, in that case, I'll respect your decision to keep it for insiders and not try to reverse engineer the code.

~~Quick question: do insiders get more features or earlier access to features?~~ Just read the page and found out they get earlier access.

If it's the second one, how long do they usually take to be released to the general public? I would love to add instant previews to part of my project.

**Reply by squidfunk** [reply]
It depends on the rate at which we get new sponsors, so it's hard to say. Instant previews are in the [22k goal](https://squidfunk.github.io/mkdocs-material/insiders/benefits/#22000-habanero-orange), so it will definitely take some time, but who knows, maybe we'll reach it in a few months or a year? ☺️ Sponsors get all features immediately.

**Reply by axololly** [reply]
Ooh, yeah, that might take some serious time. Good on you and the others for raising that much money.

What do you suggest I do in the meantime for a "it's cheaper to make it at home" solution?

**Reply by squidfunk** [reply]
There are probably libraries out there that implement this functionality, but I'm not sure what goes well with Material for MkDocs. That's the reason why we implemented it natively, as it allows us to offer much better support.

**Reply by axololly** [reply]
Understandable. Would [this](https://github.com/zachhannum/mkdocs-tooltipster-links-plugin) be an allowed solution?
